packages/@ember/-internals/glimmer/lib/renderer.ts

```diff
@@ -510,7 +525,16 @@ export abstract class Renderer {
     if (this._isValid()) {
       return;
     }
-    this._renderRootsTransaction();
+
+    if (renderSettledDeferred !== null) {
```

Comment:
We only use a `try`/`catch` when we need to manually reject the `renderSettled` promise. This allows us to ensure that the normal developer ergonomics are still good, while supporting the requirement of rejecting the `ApplicationInstance.prototype.visit` promise. 

Specifically, if we always wrapped in `try`/`catch` when you have an error thrown from a userland hook (e.g. component constructor, didInsertElement, etc) the "pause on uncaught exceptions" feature of DevTools will pause deep in the renderer internals. Without the `try`/`catch` the breakpoint you hit will actually be in userland code at the location of the error.

Reply:
Thank you @rwjblue